tests/test_misc.py

```diff
@@ -33,23 +26,21 @@ class A(metaclass=Singleton):
 
 def test_MultiDict():
     assert issubclass(MultiDict, MutableMapping)
-    x = MultiDict(None)
-    # setitem
-    x["a"] = "b"
-    # getitem
-    assert x["a"] == "b"
-    assert x.get("a") == "b"
-
-    del x["a"]
-    assert x.get("a") is None
-    assert x.__repr__() == str(x)
-    assert str(x) == "MultiDict({})"
-    x["b"] = "abc"
+    multi_dict = MultiDict({"a": ["a", "b", "c"]})
+    assert multi_dict["a"] == "c"
+    assert multi_dict.get("a") == "c"
+    assert multi_dict.getlist("a") == ["a", "b", "c"]
+
+    del multi_dict["a"]
+    assert multi_dict.get("a") is None
+    assert multi_dict.__repr__() == str(multi_dict)
+    assert str(multi_dict) == "MultiDict({})"
+    multi_dict["b"] = "abc"
     acc = 0
-    for a in x:
+    for _ in multi_dict:
```

Comment:
I'd say the dict should have a `len` method, and if we want to test the iteration, we should also check the values